qiskit/synthesis/two_qubit/xx_decompose/circuits.py

```diff
@@ -291,8 +292,8 @@ def canonical_xx_circuit(target, strength_sequence, basis_embodiments):
         circuit.compose(basis_embodiments[strength_sequence[0]], inplace=True)
         circuit.compose(source_reflection.inverse(), inplace=True)
         circuit.compose(source_shift, inplace=True)
-        circuit.global_phase += -np.log(shift_phase_shift).imag
-        circuit.global_phase += -np.log(reflection_phase_shift).imag
+        circuit.global_phase -= cmath.phase(shift_phase_shift)
+        circuit.global_phase -= cmath.phase(reflection_phase_shift)
```

Comment:
We're strongly assuming that the things marked `phase_shift` have unit magnitude, I assume?  Feels like this change could bring about a change in the exact numerics, though if we were particularly worried about that, feels like we ought to calculate with the phase shifts stored as angles directly.

I'm fine if this passes the test suite, but might cause some minor merge conflicts with #12105.

Reply:
even if not unit magnitude `cmath.phase(scalar)` and `np.log(scalar).imag` should be the same

Reply:
```ipython
In [26]: q = 100.+22.2j

In [27]: cmath.phase(q) - np.log(q).imag
Out[27]: 0.0
```

Reply:
oh yeah true sorry, I was just thinking about the logarithm in general, which is obviously nonsense